phonon/fields.py

```diff
@@ -0,0 +1,106 @@
+import itertools
+
+import phonon.connections
+
+
+class Field(object):
+
+    def key(self, *args):
+        return ".".join([str(a) for a in args])
+
+
+class ID(Field):
+
+    def __init__(self):
+        pass
+
+    def merge(self, a, b):
+        assert a == b
+        return a
+
+    def cache(self, *args):
+        return True
+
+
+class Sum(Field):
+
+    def __init__(self, dtype=int):
```

Comment:
Yeah, exactly... it's `data_type`. I chose the variable name since it will be familiar to those with experience in numpy/scipy. Perhaps I should just use `data_type`.